consensus/ibft/src/integration-test/java/tech/pegasys/pantheon/consensus/ibft/tests/RoundChangeTest.java

```diff
@@ -0,0 +1,319 @@
+import tech.pegasys.pantheon.consensus.ibft.ibftmessagedata.PreparedCertificate;
+import tech.pegasys.pantheon.consensus.ibft.ibftmessagedata.ProposalPayload;
+import tech.pegasys.pantheon.consensus.ibft.ibftmessagedata.RoundChangeCertificate;
+import tech.pegasys.pantheon.consensus.ibft.ibftmessagedata.RoundChangePayload;
+import tech.pegasys.pantheon.consensus.ibft.ibftmessagedata.SignedData;
+import tech.pegasys.pantheon.consensus.ibft.support.RoundSpecificNodeRoles;
+import tech.pegasys.pantheon.consensus.ibft.support.TestContext;
+import tech.pegasys.pantheon.consensus.ibft.support.TestContextFactory;
+import tech.pegasys.pantheon.consensus.ibft.support.ValidatorPeer;
+import tech.pegasys.pantheon.ethereum.core.Block;
+
+import java.time.Clock;
+import java.util.List;
+import java.util.Optional;
+
+import com.google.common.collect.Lists;
+import org.junit.Before;
+import org.junit.Test;
+
+public class RoundChangeTest {
+
+  private final long blockTimeStamp = 100;
+  private final Clock fixedClock =
+      Clock.fixed(Instant.ofEpochSecond(blockTimeStamp), ZoneId.systemDefault());
+
+  private final TestContext context = TestContextFactory.createTestEnvironment(4, 0, fixedClock);
+  private final ConsensusRoundIdentifier roundId = new ConsensusRoundIdentifier(1, 0);
+  private final RoundSpecificNodeRoles roles = context.getRoundSpecificRoles(roundId);
+
+  private final MessageFactory localNodeMessageFactory = context.getLocalNodeMessageFactory();
+
+  private final Block blockToPropose = context.createBlockForProposal(0, 15);
+
+  @Before
+  public void setup() {
+    context.getController().start();
+  }
+
+  @Test
+  public void onRoundChangeTimerExpiryEventRoundChangeMessageIsSent() {
+
+    // NOTE: The prepare certificate will be empty as insufficient Prepare msgs have been received.
+    final ConsensusRoundIdentifier targetRound = new ConsensusRoundIdentifier(1, 1);
+    final SignedData<RoundChangePayload> expectedTxRoundChange =
+        localNodeMessageFactory.createSignedRoundChangePayload(targetRound, empty());
+    context.getController().handleRoundExpiry(new RoundExpiry(roundId));
+    assertPeersReceivedExactly(roles.getAllPeers(), expectedTxRoundChange);
+  }
+
+  @Test
+  public void roundChangeHasEmptyCertificateIfNoPrepareMessagesReceived() {
+    final ConsensusRoundIdentifier targetRound = new ConsensusRoundIdentifier(1, 1);
+    final SignedData<RoundChangePayload> expectedTxRoundChange =
+        localNodeMessageFactory.createSignedRoundChangePayload(targetRound, empty());
+
+    roles.getProposer().injectProposal(roundId, blockToPropose);
+    roles.getAllPeers().forEach(ValidatorPeer::clearReceivedMessages);
+
+    context.getController().handleRoundExpiry(new RoundExpiry(roundId));
+    assertPeersReceivedExactly(roles.getAllPeers(), expectedTxRoundChange);
+  }
+
+  @Test
+  public void roundChangeHasPopulatedCertificateIfQuorumPrepareMessagesAreReceived() {
```

Comment:
nit: isn't this really we have quorum prepare messages and a proposal. without the proposal there wouldn't be a prepared cert if I remember correctly.

Reply:
updated.